pkg/azureclient/client.go

```diff
@@ -155,23 +156,26 @@ func newClient(authJSONSource func() ([]byte, error)) (*azureClient, error) {
 		return nil, errors.New("missing subscriptionId in auth")
 	}
 
-	config := auth.NewClientCredentialsConfig(clientID, clientSecret, tenantID)
+	env, err := azure.EnvironmentFromName(string(environmentName))
```

Comment:
can we do the common `if environmentName == "" { environmentName = azure.PublicCloud.Name }` dance here instead of sprinkled all around the various files?